migration/src/spreadsheets.rs

```diff
@@ -130,6 +132,68 @@ impl SheetResult {
         })
     }
 
+    /// Parse this sheet as the collection index. Updates a Collection 
+    /// which contains two Chapter Groups, each of which 
+    /// has a list of associated Chapters.
+    pub fn into_collection_index(self,
+        self_title: &String, 
+        self_wordpress_menu_id: &i64) -> Result<Collection> {
+        let mut chapters = Vec::new();
+        let mut self_intro_chapters = Vec::new();
+        for row in self.values.into_iter().skip(2){
+            if row[0].is_empty(){
+                self_intro_chapters = chapters.clone();
+                chapters = Vec::<Chapter>::new();
+            }
+            else {
+                let index_in_parent_string = row[0].clone();
+                let index_i64 = index_in_parent_string.parse::<i64>().unwrap();
+
+                let document_if_applicable = row[3].clone();
+                let mut doc_string= None;
+
+                let wp_id_if_applicable = row[5].clone();
+                let mut wp_id = None;
+
+                if wp_id_if_applicable.is_empty(){
+                    wp_id = None;
+                }
+                else {
+                    wp_id = Some(wp_id_if_applicable.parse::<i64>().unwrap());
```

Comment:
Avoid using `.unwrap()` whenever possible because if you unwrap a `None` it will crash the whole program (panic) instead of returning an `Error` from this function. You can do `Some(x.parse()?)` (if you want a failed parse to return an error) or `x.parse().ok()` (if you want a failed parse to result in no value for `wp_id`).

Reply:
Cool, fixed this, and used this logic to remove the if/else statements, thanks!